packages/tools/kda-cli/src/questions/deploy.ts

```diff
@@ -0,0 +1,227 @@
+  if (!Array.isArray(deployTargets)) return false;
+  if (deployTargets?.includes('l1')) return true;
+  if (deployTargets?.includes('l1l2')) return true;
+  return false;
+};
+
+const isL2 = ({ deployTargets }: IAnswers): boolean => {
+  if (!Array.isArray(deployTargets)) return false;
+  if (deployTargets?.includes('l2')) return true;
+  if (deployTargets?.includes('l1l2')) return true;
+  return false;
+};
+const signAndSend = ({
+  publicKey,
+  secretKey,
+  useChainWeaver,
+}: {
+  useChainWeaver: boolean;
+  publicKey: string;
+  secretKey: string;
+}) => {
+  if (useChainWeaver) return buildCommand(signWithChainweaver, send);
+  return buildCommand(signWithKeypair({ publicKey, secretKey }), send);
+};
+
+const deploy = async ({
+  chainIds,
+  setDeploySettings,
+  setDeployChainSettings,
+}: {
+  chainIds: string[];
+  setDeployChainSettings: (chainId: string) => Reducer;
+  setDeploySettings: Reducer;
+}): Promise<unknown> => {
+  try {
+    for (const chainId of chainIds) {
+      const res = await buildCommand(
+        setDeployChainSettings(chainId),
+        setDeploySettings,
+      )({});
+      console.log('res', res);
+    }
+    return { success: true };
+  } catch (e) {
+    console.log('e', e);
+    return { success: false };
+  }
+};
+const setDeployChainSettings =
+  ({
+    signer,
+    network,
+    endpoint,
+  }: {
+    signer: string;
+    network: string;
+    endpoint: string;
+  }) =>
+  (chainId: string): Reducer =>
+    buildCommand(
+      setMeta({
+        gasLimit: 100000,
+        gasPrice: 0.0000001,
+        ttl: 60000,
+        chainId: chainId,
+        sender: signer,
+      }),
+      setNetworkId(network),
+      setDomain(endpoint),
+    );
+const setDeploySettings = ({
+  pactCode,
+  data,
+  publicKey,
+  secretKey,
+  useChainWeaver,
+}: {
+  pactCode: string;
+  publicKey: string;
+  secretKey: string;
+  useChainWeaver: boolean;
+  data: unknown;
+}) =>
+  buildCommand(
+    setCommand(pactCode),
+    addCapability({
+      name: 'UNRESTRICTED',
+      args: [],
+      signer: publicKey,
+    }),
+    setData(data),
+    signAndSend({ publicKey, secretKey, useChainWeaver }),
+    listen,
+  );
+
+export const deployQuestions: IQuestion[] = [
+  {
+    message: 'What would you like to deploy? (Provide a `.pact` file)',
+    name: 'pactFile',
+    type: 'input',
+    when: isDeployTask,
+  },
+  {
+    message: 'What data goes with it? (Provide a `.json` file)',
+    name: 'dataFile',
+    type: 'input',
+    when: isDeployTask,
+  },
+  {
+    message: 'Where would you like to deploy?',
+    name: 'deployTargets',
+    type: 'multi-select',
+    choices: [
+      { label: 'L1', value: 'l1' },
+      { label: 'L2', value: 'l2' },
+      { label: 'L1 & L2', value: 'l1l2' },
+    ],
+    when: isDeployTask,
+  },
+  {
+    message: 'On which chains on L1 would you like to deploy?',
+    name: 'l1Chains',
+    type: 'multi-select',
+    choices: [...Array(20).keys()].map((i) => ({
```

Comment:
We might want to have this configurable

Minor nitpick: I like this more :P (idem for other occurrences)

```suggestion
    choices: Array.from(Array(20)).map((i) => ({
```

Reply:
Regarding the configurable part, since we don't expect the amount of chains to change any time soon, I'll leave it at this for now. But if we need to update this in the future, we should indeed make this configurable